staging/src/k8s.io/apiserver/pkg/storage/names/generate.go

```diff
@@ -42,7 +42,7 @@ var SimpleNameGenerator NameGenerator = simpleNameGenerator{}
 const (
 	// TODO: make this flexible for non-core resources with alternate naming rules.
 	maxNameLength          = 63
-	randomLength           = 5
+	randomLength           = 10
```

Comment:
LGTM, let's double check that @liggitt also can't think of how this could break anyone?

Reply:
I mean... I wouldn't endorse this, but https://github.com/kubesphere/kubesphere/blob/master/pkg/simple/client/logging/elasticsearch/elasticsearch.go#L284 was the first result for https://grep.app/search?q=%5D%7B5%7D&filter[lang][0]=Go

Reply:
will it break some people? yes.

are those people relying on unspecified behavior? yes.

is it worth it? I don't know? is this fixing a hypothetical conflict issue or one people are actually encountering?

Reply:
That's a pretty weird thing to do when you could just look at the controller reference.

The purpose of this is to substantially reduce the number of times generateName fails because it chooses a dup.

Reply:
It's unclear if collisions ACTUALLY happen, which is why the original discussion was about retrying create with a different random suffix.  I'd feel better if we had a much larger space than we do today, but I'm not sure it's worth breaking.

I filed that as a bug.  I don't know why I am surprised to find people doing that, even though this is *literally* the reason labels exist.  https://github.com/kubesphere/kubesphere/issues/5584

Digging into others, well, there are more.

Reply:
> unclear if collisions ACTUALLY happen

Collisions actually happen, I vaguely recall 2 reports? They are rare AFAIK.

Reply:
https://github.com/gocrane/crane/issues/716

https://github.com/rancher/rancher/issues/40852

https://github.com/weaveworks/wksctl/issues/342

https://github.com/telepresenceio/telepresence/issues/3063

https://github.com/hashicorp/terraform-provider-kubernetes/issues/2037

Reply:
Given this, I am disinclined to pursue the change.  Maybe we can yell about it and come back in a release or two.

Reply:
That's wait too many projects that would break, even though it's on them.

I noticed that `0` and `1` are not part of `alphanums`. Is it because they look too much like vowels? I would take the risk.

Reply:
I kind of like having short random strings because it allows me to write `k get pod name-xxxxx` without having to reach for my mouse to copy-paste a pod name :sweat_smile: 

Adding a couple of characters to the list would get us to 20M possibilities.

Reply:
Adding 1 or 100 characters is the same problem.

Adding to the character set has very minimal impact on p(collision) and has its own risk.  Some of those projects embed the valid charset (sigh), but we will inevitable end up with a pod named "foobar-ac0ck".